src/Microsoft.AspNetCore.StaticFiles/StaticFileContext.cs

```diff
@@ -425,5 +433,63 @@ private ContentRangeHeaderValue ComputeContentRange(RangeItemHeaderValue range,
             length = end - start + 1;
             return new ContentRangeHeaderValue(start, end, _length);
         }
+
+        //internal for testing
+        internal static void ApplyCacheProfile(HttpContext context, CacheProfile cacheProfile)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+            if (cacheProfile == null)
+            {
+                throw new ArgumentNullException(nameof(cacheProfile));
+            }
+
+            var headers = context.Response.Headers;
+
+            if (!string.IsNullOrEmpty(cacheProfile.VaryByHeader))
+            {
+                headers[HeaderNames.Vary] = cacheProfile.VaryByHeader;
+            }
+
+            if (cacheProfile.NoStore)
+            {
+                headers[HeaderNames.CacheControl] = "no-store";
+
+                // Cache-control: no-store, no-cache is valid.
+                if (cacheProfile.Location == ResponseCacheLocation.None)
+                {
+                    headers.AppendCommaSeparatedValues(HeaderNames.CacheControl, "no-cache");
+                    headers[HeaderNames.Pragma] = "no-cache";
+                }
+            }
+            else
+            {
+                string cacheControlValue = null;
+                switch (cacheProfile.Location)
+                {
+                    case ResponseCacheLocation.Any:
+                        cacheControlValue = "public";
+                        break;
+                    case ResponseCacheLocation.Client:
+                        cacheControlValue = "private";
+                        break;
+                    case ResponseCacheLocation.None:
+                        cacheControlValue = "no-cache";
+                        headers[HeaderNames.Pragma] = "no-cache";
+                        break;
+
+                    default:
+                        var exception = new NotImplementedException($"Unknown {nameof(ResponseCacheLocation)}: {cacheProfile.Location}");
+                        Debug.Fail(exception.ToString());
+                        throw exception;
+                }
+
+                cacheControlValue = cacheControlValue + ",max-age=" + (int)cacheProfile.Duration.TotalSeconds;
```

Comment:
I thought I was testing incorrectly so I removed my comment but it seems like there is actually a very slight increase in perf when placing this concatenation inside the switch statement. The difference is pretty small but it might be worth it.